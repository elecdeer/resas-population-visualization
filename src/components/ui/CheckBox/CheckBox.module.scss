.checkBox {
	cursor: pointer;
	.input {
		cursor: pointer;
	}
	.label {
		user-select: none;
	}
}