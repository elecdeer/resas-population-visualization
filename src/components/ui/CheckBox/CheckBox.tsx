import React from "react";
import styles from "./CheckBox.module.scss";

export type CheckBoxProps = {
  isChecked?: boolean;
  defaultChecked?: boolean;
  onChange?: React.ChangeEventHandler<HTMLInputElement>;
};

export const CheckBox: React.FC<CheckBoxProps> = ({
  isChecked,
  defaultChecked,
  onChange,
  children,
}) => {
  return (
    <label className={styles.checkBox}>
      <input
        className={styles.input}
        type={"checkbox"}
        defaultChecked={defaultChecked}
        checked={isChecked}
        onChange={onChange}
      />
      <span className={styles.label}>{children}</span>
    </label>
  );
};
