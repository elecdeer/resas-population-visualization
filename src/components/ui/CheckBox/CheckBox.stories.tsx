import { CheckBox, CheckBoxProps } from "./CheckBox";

export default {
  title: "ui/CheckBox",
  component: CheckBox,
};

export const Default = (args: CheckBoxProps) => <CheckBox {...args} />;

export const Labeled = (args: CheckBoxProps) => (
  <CheckBox {...args} defaultChecked={true}>
    Label
  </CheckBox>
);
