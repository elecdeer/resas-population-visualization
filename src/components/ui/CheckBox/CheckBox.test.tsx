import { CheckBox } from "./CheckBox";
import { render } from "@testing-library/react";

test("CheckBox Component Snapshot Test", () => {
  const { asFragment } = render(<CheckBox />);
  expect(asFragment()).toMatchSnapshot();
});
