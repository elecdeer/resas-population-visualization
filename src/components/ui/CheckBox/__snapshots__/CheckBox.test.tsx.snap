// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox Component Snapshot Test 1`] = `
<DocumentFragment>
  <label
    class="checkBox"
  >
    <input
      class="input"
      type="checkbox"
    />
    <span
      class="label"
    />
  </label>
</DocumentFragment>
`;
